### ISSUE TITLE
VAAPI segmentation fault on playlist next

### ISSUE THREAD

**Issue by hectorBrown** [mpv-player/mpv#11698]
### Important Information

Provide following Information:
- mpv version
0.35.1-dirty
- Linux Distribution and Version
Arch Linux on 6.1.29-1-lts
- Source of the mpv binary
[Package](https://archlinux.org/packages/community/x86_64/mpv/) -- version: mpv-1:0.35.1-6
- If known which version of mpv introduced the problem
Not known.
- Window Manager and version
i3 4.22
- GPU driver and version
Mesa 23.0.3
- Possible screenshot or video of visual glitches
N/A

### Reproduction steps

1. `ls [some folder of videos] | mpv --playlist=- --no-config --hwdec=vaapi --interpolation --video-sync=display-resample`
2. Use `>` to access the next video in the playlist and the crash should occur.

Specifically I only tested .mkvs, but occurs with both .h264 and .h265.
Any 2 of the specified options seem to work fine, only all 3 in tandem cause the issue as far as I can tell.

### Expected behavior

The next video should play as normal.

### Actual behavior

Video does not play, mpv crashes citing a "segmentation fault"

### Log file

[mpv.log](https://github.com/mpv-player/mpv/files/11511929/mpv.log)

### Sample files

N/A

**Comment by Traneptora** [comment]
I can't reproduce this, although I have an NVidia card, not an AMD card. What happens if you use --vo=gpu-next?

**Comment by hectorBrown** [comment]
Ah, yeah, that fixes it.

**Comment by quietvoid** [comment]
I also can't reproduce this with a discrete GPU (AMD Radeon 6900 XT)

**Comment by sfan5** [comment]
Can't reproduce on my machine with Linux 6.3.2-arch1-1, mpv acababec20 and Radeon 6500 XT.

**Comment by hectorBrown** [comment]
I'm on AMD Ryzen 5 3500U with Radeon Vega 8 graphics.

**Comment by HxxThSwggr** [comment]
Just started happening today on my setup after I updated to [this](https://github.com/mpv-player/mpv/commit/e8126e54e51d0ed8c052bf8733df5519d003e464).
Video crashes after a minute or so of playback (once it crashed after a second)
playlists have no effect
`hwdec = no` has no effect
happens on both 264 and 265

**Also AMD integrated (Ryzen 5 5600G)**
I believe there weren't any `mesa` updates that could break something

Few day ago I ran mpv just fine. Idk exactly what commit but [this](https://github.com/mpv-player/mpv/commit/2dc8e70093d9f80a01696f059ff02784bcb7b5f0) one at least.
